src/core/DataFile.cpp

```diff
@@ -1386,6 +1387,43 @@ void DataFile::upgrade_noHiddenClipNames()
 }
 
 
+/** \brief Note range has been extended to match MIDI specification
+ *
+ * Default base note number changed from 57 to 69. To ensure that patterns
+ * copied to a new instrument instance sound correctly, the adjusment is
+ * are adjusted instead of just shifting the instrument base note.
```

Comment:
Also, I don't understand what you mean here. The default base pitch for new instruments is now 69, and users will see that notes in patterns as well as base notes are shifted down by an octave.

Reply:
Thanks for spotting the typo. I will update the comment to explain better what I meant.

The upgrade routine must ensure that instruments in existing projects "stay broken". Previously the routine simply moved the base note an octave higher, so that the instrument keeps sounding "an octave too low", therefore preserving how it sounds in the project.

But as Dom mentioned in his review, if, for example, the user opens a project and copies a pattern from old ZynAddSubFX instance to a newly added ZynAddSubFX instance, the pattern will sound an octave higher in the new instance, which is not expected. So instead of lowering the pitch by moving the base note up, the individual notes in all patterns are moved an octave lower to achieve the same effect while improving the pattern copy behavior.

Reply:
Yes, but then clips for normal instruments like TripleOscillator will sound an octave lower when copied. I will com with a proposal soon.

Reply:
So I did do a simple "case study":

There are two instruments in a project created in 1.2.2, a TripleOsc and Zyn. Both of them have the base note set to A4 = 57, and they both have one pattern with one A4 note. When the base note or the pattern is played, TriplOsc will produce a sound matching A4 (440 Hz) and Zyn will produce a sound matching A3 (220 Hz).

The current upgrade routine lowers all notes in patterns assigned to Zyn, but at the same time the key mapping and note definition has changed and A4 now equals 69. So the result will be: TripleOsc has now base note set to A3 = 57, the note in its assigned pattern says "A3", but it plays A4 (440 Hz) and when copied to a new instance the pattern will play A3. With Zyn the base note is 57, the note in its pattern becomes "A2" and plays "A3" (220 Hz). When copied to another (new) Zyn instance it will sound "true" A2 (110 Hz). So the upgrade routine actually does not even work as intended for the affected subset of instruments.

What I think needs to happen is: 1) move the base note of _all_ instruments from 57 to 69 so that the instruments really play the note that "it says on the screen" (unless base note has been already moved before); 2) move the notes in all the patterns assigned to instruments that were _not_ affected by the "midi offset" bug one octave higher. This will ensure that all the instruments have correctly labeled notes which will sound as intended when copied, and leave the notes of affected MIDI-based instruments sounding an octave lower.

Now after some thinking it seems really easy and obvious...

Reply:
Your idea is the same as I thought, LGTM